sphinx/writers/latex.py

```diff
@@ -1339,7 +1339,14 @@ def visit_caption(self, node: Element) -> None:
         elif self.table and node.parent.tagname == 'figure':
             self.body.append('\\sphinxfigcaption{')
         else:
-            self.body.append('\\caption{')
+            # Use alt text as short caption for the \listoffigures
+            short = ''
+            if isinstance(node.parent, nodes.figure):
+                if isinstance(node.parent.children[0], nodes.image):
+                    alt = node.parent.children[0].get('alt')
+                    if alt:
+                        short = '[' + texescape.escape(alt) + ']'
```

Comment:
How about this? I think this condition should be flatten as others above.
```suggestion
    elif (isinstance(node.parent, node.figure) and isinstance(node.parent.children[0], node.image) and 'alt' in node.parent.children[0]):
        alt = node.parent.children[0]['alt']
        self.body.append(r'\caption[%s]{' % texescape.escape(alt))
    else:
```